Sensors/SMCDellSensors/SMCDellSensors.hpp

```diff
@@ -0,0 +1,146 @@
+//
+//  SMCDellSensors.hpp
+//  SMCDellSensors
+//
+//  Copyright (C) 2001  Massimo Dal Zotto <dz@debian.org>
+//  http://www.debian.org/~dz/i8k/
+//  more work https://www.diefer.de/i8kfan/index.html , 2007
+//
+//  SMIMonitor ported from FakeSMC plugin (created by Slice 2014).
+//
+
+#ifndef SMCDellSensors_hpp
+#define SMCDellSensors_hpp
+
+#include "SMIMonitor.hpp"
+#include <Library/LegacyIOService.h>
+#include <IOKit/IOReportTypes.h>
+#pragma clang diagnostic push
+#pragma clang diagnostic ignored "-Winconsistent-missing-override"
+#include <IOKit/acpi/IOACPIPlatformDevice.h>
+#pragma clang diagnostic pop
+#include <Sensors/Private/pwr_mgt/IOPMPowerSource.h>
+#include <Sensors/Private/pwr_mgt/RootDomain.h>
+#include <IOKit/IOTimerEventSource.h>
+
+#include <Headers/kern_util.hpp>
+
+class EXPORT SMCDellSensors : public IOService {
+	OSDeclareDefaultStructors(SMCDellSensors)
+
+	/**
+	 *  Key name index mapping
+	 */
+	static constexpr size_t MaxIndexCount = sizeof("0123456789ABCDEF") - 1;
+	static constexpr const char *KeyIndexes = "0123456789ABCDEF";
+
+	/**
+	 *  Key name declarations
+	 */
+	
+	static constexpr SMC_KEY KeyFNum = SMC_MAKE_IDENTIFIER('F','N','u','m');
+	static constexpr SMC_KEY KeyF0ID(size_t i) { return SMC_MAKE_IDENTIFIER('F',KeyIndexes[i],'I','D'); }
+	static constexpr SMC_KEY KeyF0Ac(size_t i) { return SMC_MAKE_IDENTIFIER('F',KeyIndexes[i],'A','c'); }
+	static constexpr SMC_KEY KeyF0Mn(size_t i) { return SMC_MAKE_IDENTIFIER('F',KeyIndexes[i],'M','n'); }
+	static constexpr SMC_KEY KeyF0Mx(size_t i) { return SMC_MAKE_IDENTIFIER('F',KeyIndexes[i],'M','x'); }
+	static constexpr SMC_KEY KeyF0Md(size_t i) { return SMC_MAKE_IDENTIFIER('F',KeyIndexes[i],'M','d'); }
+	static constexpr SMC_KEY KeyF0Tg(size_t i) { return SMC_MAKE_IDENTIFIER('F',KeyIndexes[i],'T','g'); }
+	// the following constant defines smc key 'FS! '
+	static constexpr SMC_KEY KeyFS__ = SMC_MAKE_IDENTIFIER('F','S','!',' ');
+	
+	static constexpr SMC_KEY KeyTC0P(size_t i) { return SMC_MAKE_IDENTIFIER('T','C',KeyIndexes[i],'P'); }
+	static constexpr SMC_KEY KeyTG0P(size_t i) { return SMC_MAKE_IDENTIFIER('T','G',KeyIndexes[i],'P'); }
+	static constexpr SMC_KEY KeyTm0P(size_t i) { return SMC_MAKE_IDENTIFIER('T','m',KeyIndexes[i],'P'); }
+	static constexpr SMC_KEY KeyTN0P(size_t i) { return SMC_MAKE_IDENTIFIER('T','N',KeyIndexes[i],'P'); }
+	static constexpr SMC_KEY KeyTA0P(size_t i) { return SMC_MAKE_IDENTIFIER('T','A',KeyIndexes[i],'P'); }
+	static constexpr SMC_KEY KeyTW0P(size_t i) { return SMC_MAKE_IDENTIFIER('T','W',KeyIndexes[i],'P'); }
+	
+	/**
+	 *  VirtualSMC service registration notifier
+	 */
+	IONotifier *vsmcNotifier {nullptr};
+
+	/**
+	 *  Registered plugin instance
+	 */
+	VirtualSMCAPI::Plugin vsmcPlugin {
+		xStringify(PRODUCT_NAME),
+		parseModuleVersion(xStringify(MODULE_VERSION)),
```

Comment:
Please change the module version to from 1.0 to 1.0.0. Currently there is a buffer overflow in `parseModuleVersion`.